src/model/namespace/NamespaceId.ts

```diff
@@ -38,23 +40,31 @@ export class NamespaceId {
      * Create NamespaceId from namespace string name (ex: nem or domain.subdom.subdome)
      * or id in form of array number (ex: [929036875, 2226345261])
      *
-     * @param id
+     * @param   {string|number[]}   id          The namespace id (string or uint64 array)
+     * @param   {NetworkType}       networkType The network type for hash algorithm resolution
      */
-    constructor(id: string | number[]) {
+    constructor(
+        id: string | number[],
+        networkType?: NetworkType,
```

Comment:
should remove the `?` I think

Reply:
not its intended ; using `!` in else block. 

When you instantiate a NamespaceId by number[] you don't need any NetworkType and would be wrong to default it _then_ to something.

When you instantiate by string, you always need the networkType.